src/ring/middleware/webjars.clj

```diff
@@ -1,20 +1,27 @@
 (ns ring.middleware.webjars
   (:import [org.webjars WebJarAssetLocator])
-  (:require [ring.middleware.head :as head]
+  (:require [clojure.string :as string]
+            [ring.middleware.head :as head]
             [ring.util.codec :as codec]
             [ring.util.request :as req]
             [ring.util.response :as resp]))
 
 (def ^:private webjars-pattern
   #"META-INF/resources/webjars/([^/]+)/([^/]+)/(.*)")
 
-(defn- asset-path [prefix resource]
-  (let [[_ name version path] (re-matches webjars-pattern resource)]
-    (str prefix "/" name "/" path)))
+(defn- path [& parts]
+  (string/join "/" parts))
+
+(defn- asset-paths [prefix resource]
+  (let [[_ name version filename] (re-matches webjars-pattern resource)
+        path-with-version    (path prefix name version filename)
+        path-without-version (path prefix name filename)]
+    {path-with-version    resource
+     path-without-version resource}))
```

Comment:
This could just be:

```clojure
{(str prefix "/" name "/" version "/" filename) resource
 (str prefix "/" name "/" filename)             resource}
```